crates/uv/src/commands/project/version.rs

```diff
@@ -0,0 +1,593 @@
+            python,
+            install_mirrors,
+            &settings,
+            network_settings,
+            python_preference,
+            python_downloads,
+            concurrency,
+            no_config,
+            cache,
+            short,
+            output_format,
+            printer,
+            preview,
+        ))
+        .await;
+    }
+
+    let old_version = toml.version().map_err(|err| match err {
+        Error::MalformedWorkspace => {
+            if toml.has_dynamic_version() {
+                anyhow!(
+                    "We cannot get or set dynamic project versions in: {}",
+                    pyproject_path.user_display()
+                )
+            } else {
+                anyhow!(
+                    "There is no 'project.version' field in: {}",
+                    pyproject_path.user_display()
+                )
+            }
+        }
+        err => {
+            anyhow!("{err}: {}", pyproject_path.user_display())
+        }
+    })?;
+
+    // Figure out new metadata
+    let new_version = if let Some(value) = value {
+        match Version::from_str(&value) {
+            Ok(version) => Some(version),
+            Err(err) => match &*value {
+                "major" | "minor" | "patch" => {
+                    return Err(anyhow!(
+                        "Invalid version `{value}`, did you mean to pass `--bump {value}`?"
+                    ));
+                }
+                _ => {
+                    return Err(err)?;
+                }
+            },
+        }
+    } else if let Some(bump) = bump {
+        Some(bumped_version(&old_version, bump, printer)?)
+    } else {
+        None
+    };
+
+    // Update the toml and lock
+    let status = if dry_run {
+        ExitStatus::Success
+    } else if let Some(new_version) = &new_version {
+        let project = update_project(project, new_version, &mut toml, &pyproject_path)?;
+        Box::pin(lock_and_sync(
+            project,
+            project_dir,
+            locked,
+            frozen,
+            active,
+            no_sync,
+            python,
+            install_mirrors,
+            &settings,
+            network_settings,
+            python_preference,
+            python_downloads,
+            installer_metadata,
+            concurrency,
+            no_config,
+            cache,
+            printer,
+            preview,
+        ))
+        .await?
+    } else {
+        debug!("No changes to version; skipping update");
+        ExitStatus::Success
+    };
+
+    // Report the results
+    let old_version = VersionInfo::new(name.as_ref(), &old_version);
+    let new_version = new_version.map(|version| VersionInfo::new(name.as_ref(), &version));
+    print_version(old_version, new_version, short, output_format, printer)?;
+
+    Ok(status)
+}
+
+/// Find the pyproject.toml we're modifying
+///
+/// Note that `uv version` never needs to support PEP723 scripts, as those are unversioned.
```

Comment:
Nit: This is the proper style
```suggestion
/// Note that `uv version` never needs to support PEP 723 scripts, as those are unversioned.
```